### PR TITLE
ENH Convert ElementList into a functional component

### PR DESCRIPTION
**Pull request by GuySartorelli**
This is needed to allow the use of hooks for dnd-kit.

This PR should change nothing about the user experience or functionality of the module, it should be a perfect like-for-like comparison to `6.0` without this PR.

Test everything you can think to test - sorting, saving, publishing, archiving, etc both from a page and an element level to check everything works the way it used to.

## Issue
- https://github.com/silverstripe/.github/issues/343

### CODE REVIEW

**Comment by GuySartorelli** [comment]
behat failure will be fixed by https://github.com/silverstripe/silverstripe-elemental/pull/1296 separately

File: client/src/legacy/ElementEditor/entwine.js [hunk 1]
```diff
@@ -116,12 +117,10 @@ jQuery.entwine('ss', ($) => {
...
         this.setIncrement(increment);
         sharedObject.entwineResolve = entwineResolve;
-        // setState() is bound in the constructor of the ElementList react component
-        // setting saveAllElementst to true will trigger a re-render in the react component
-        sharedObject.setState({
-          saveAllElements: true,
-          increment
-        });
+        // These state setting methods are set by the ElementList react component
+        // setting saveAllElementst state to true will trigger a re-render in the react component
```
Comment (emteknetnz) [inline client/src/legacy/ElementEditor/entwine.js]:
```suggestion
        // setting saveAllElements state to true will trigger a re-render in the react component
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]
It appears to do the things correctly

One tiny change

File: client/src/legacy/ElementEditor/entwine.js (hunk 1 above)
Comment (GuySartorelli) [reply client/src/legacy/ElementEditor/entwine.js]: done

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**